### ISSUE TITLE
page.update() after page.views.clear() sets page to None.

### ISSUE THREAD

**Issue by timbolshisnkykryon** [flet-dev/flet#1272]
### Description

After updating to the new version of flet (0.5.0 / 0.5.1) an existing function has stopped working. The function sets the main View object of my app, therefore, I am using page.views.clear() beforehand, in order to clear the default view. But after appending my own View and trying to update the page object, it becomes None and unusable.

### Code example to reproduce the issue:

```python
import flet as ft
from flet import (UserControl, Page, Text, AppBar, View, Column)

class MyCoolApp(UserControl):
    def __init__(self, page: Page):
        super().__init__()
        self.page = page
        self.appbar = AppBar(title=Text("Bug Report"))

    def build(self):
        self.layout = Text("Bug Report")
        return self.layout

    def initialize(self):
        self.page.views.clear()  # Works without this line
        self.page.views.append(
            View("/", [self.appbar, self.layout])
        )
        self.page.update()  # Sets page to None
        self.page.go("/")  # The error pops here when page is None

if __name__ == "__main__":
    def main(page: Page):
        page.title = "Bug Report"
        app = MyCoolApp(page)
        page.add(app)
        page.update()
        app.initialize()  # Fails at this function

    ft.app(target=main)
```

### Describe the results you received:

When the app just starts, I receive: ```AttributeError: 'NoneType' object has no attribute 'go'```

### Describe the results you expected:

No error, and a basic View displayed in my app.

### Additional information you deem important (e.g. issue happens only occasionally):

Worked before flet upgrade (0.4.2). And I fixed it for now by removing `self.page.view.clear()`.

### Flet version:

`flet==0.5.1 / 0.5.0`

### Operating system:

Windows 10 19045

**Comment by FeodorFitsner** [comment]
I'm not sure whatever you do here is correct. When you call `page.add(app)` it's basically the same as `page.views[0].append(app)` - anything you add to a "page" is added to a default "root" view. Then inside control's `initialize()` method you are clearing views collection having that control, i.e. cutting the branch you are sitting on.

What are you trying to achieve?

**Comment by FeodorFitsner** [comment]
The issue has been fixed and fix will be available in Flet 0.5.2.
You can try the fix right now by upgrading to pre-release version of Flet: `pip install flet --pre --upgrade`.
Let me know how it worked.

**Comment by timbolshisnkykryon** [comment]
Thank you @FeodorFitsner . 
Also thanks for clearing out `page.add` for me, learned from it. 
Keep up the excellent work on `flet', it is very much appreciated by the community and me.
